axlearn/common/host_array_test.py

```diff
@@ -28,26 +29,35 @@ def is_supported(
     global_batch_size: int,
     data_partition: DataPartitionType,
 ):
-    return (
-        is_supported_platform(platform)
-        and is_supported_mesh_shape(mesh_shape)
-        and (
-            data_partition == DataPartitionType.REPLICATED
-            or global_batch_size % jax.device_count() == 0
+    # TODO: consider aggregating skip reasons to efficiently root cause
+    if not is_supported_platform(platform):
+        return False, f'Platform "{platform}" not supported with devices {jax.devices()}.'
+    mesh_support, reason = is_supported_mesh_shape(mesh_shape)
+    if not mesh_support:
+        pytest.skip(reason)
```

Comment:
For consistency I'd still add dummy return statement here since the function expected to return a tuple.
```
return False, reason
```